### PR TITLE
extract protobufs into their own lib

### PR DESCRIPTION
**Pull request by nlandolfi**
For example, rather than generating a file in the data package. we generate files in the data/proto package.

Extracted the protobuf files into github.com/elos/protos, which is now a submodule.

### CODE REVIEW

**Comment by nlandolfi** [comment]
other changes leaked in

added a dev image in images/

added a Gopkg lock file from the `go dep` tool

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "protos"]
+	path = protos
+	url = https://github.com/elos/protos
```
Comment (viggyfresh) [inline .gitmodules]: dig it

File: Gopkg.lock [hunk 2]
```diff
@@ -0,0 +1,121 @@
+# This file is autogenerated, do not edit; changes may be undone by the next 'dep ensure'.
```
Comment (viggyfresh) [inline Gopkg.lock]: EXCELLENT.

File: Gopkg.toml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+[[constraint]]
+  branch = "master"
+  name = "github.com/elos/data"
+
+[[constraint]]
+  branch = "master"
+  name = "github.com/elos/grpctest"
+
+[[constraint]]
+  branch = "master"
```
Comment (viggyfresh) [inline Gopkg.toml]: Eventually, these will become releases I presume

**Comment by nlandolfi** [comment]
You are correct. And also re your comment, yes if there were releases. In
fact I'd like to fix these at shas for now prob anyway rather than a moving
target branch like master

**Pull request merged**